src/map/Popup.tsx

```diff
@@ -4,14 +4,18 @@ import styles from './Popup.module.css'
 import { Coordinate, QueryPoint, QueryPointType } from '@/stores/QueryStore'
 import Dispatcher from '@/stores/Dispatcher'
 import { AddPoint, SetPoint, SetViewportToPoint } from '@/actions/Actions'
+import { RouteStoreState } from '@/stores/RouteStore'
+import { findNextWayPoint } from '@/map/findNextWayPoint'
 
 export function PopupComponent({
     coordinate,
     queryPoints,
+    route,
     onSelect,
 }: {
     coordinate: Coordinate
     queryPoints: QueryPoint[]
+    route: RouteStoreState
```

Comment:
You could move that locig into the store, but I don't know whether this would improve things. I guess you could also pass a subset of properties instead of the whole state, but this would not be such a big difference I guess.

Reply:
Ok, let's merge this then so we can move on :)